From 37d507b3bb485589343f5d2408441fadc6e964e4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rafa=C3=ABl=20Carr=C3=A9?= <funman@videolan.org>
Date: Sat, 5 Apr 2014 11:24:55 +0200
Subject: [PATCH] strtok_r: fix detection on Windows

Make sure HAVE_STRTOK_R is always defined, to 0 if absent
Fix typo in #if
mingw-w64's pthread.h #defines strtok_r, make sure to undef it
before we use our own definition.
---
 source/common/CMakeLists.txt | 3 +++
 source/common/param.cpp      | 3 ++-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/source/common/CMakeLists.txt b/source/common/CMakeLists.txt
index 09029ba..63cdf09 100644
--- a/source/common/CMakeLists.txt
+++ b/source/common/CMakeLists.txt
@@ -130,6 +130,9 @@ if(ENABLE_ASSEMBLY)
 endif(ENABLE_ASSEMBLY)
 
 check_symbol_exists(strtok_r "string.h" HAVE_STRTOK_R)
+if(HAVE_STRTOK_R STREQUAL "")
+    set(HAVE_STRTOK_R "0")
+endif()
 set_source_files_properties(param.cpp PROPERTIES COMPILE_FLAGS -DHAVE_STRTOK_R=${HAVE_STRTOK_R})
 
 if(GCC AND GCC_HAS_NO_NARROWING)
diff --git a/source/common/param.cpp b/source/common/param.cpp
index 57e3388..ac19c7e 100644
--- a/source/common/param.cpp
+++ b/source/common/param.cpp
@@ -37,7 +37,7 @@
 #define strcasecmp _stricmp 
 #endif
 
-#if !HAVE_STROTOK_R
+#if !HAVE_STRTOK_R
 /* 
  * adapted from public domain strtok_r() by Charlie Gordon
  *
@@ -49,6 +49,7 @@
  *      http://groups.google.com/group/comp.lang.c/msg/7c7b39328fefab9c
  */
 
+#undef strtok_r
 char* strtok_r(
     char *str, 
     const char *delim, 
-- 
1.9.1

